shell/browser/extensions/extension_tab_util.h

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SHELL_BROWSER_EXTENSIONS_EXTENSION_TAB_UTIL_H_
+#define SHELL_BROWSER_EXTENSIONS_EXTENSION_TAB_UTIL_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/callback.h"
+#include "extensions/common/features/feature.h"
+#include "shell/common/extensions/api/tabs.h"
+#include "ui/base/window_open_disposition.h"
+
+class GURL;
+class ExtensionFunction;
+
+namespace electron {
+namespace api {
+class ChromeTabDetails;
+class WebContents;
+}  // namespace api
+
+}  // namespace electron
+
+namespace base {
+class DictionaryValue;
+class ListValue;
+}  // namespace base
+
+namespace content {
+class BrowserContext;
+class WebContents;
+}  // namespace content
+
+namespace gfx {
+class Rect;
+}
+
+namespace extensions {
+class Extension;
+
+// Provides various utility functions that help manipulate tabs.
+class ExtensionTabUtil {
+ public:
+  enum PopulateTabBehavior {
+    kPopulateTabs,
+    kDontPopulateTabs,
+  };
+
+  enum ScrubTabBehaviorType {
+    kScrubTabFully,
+    kScrubTabUrlToOrigin,
+    kDontScrubTab,
+  };
+
+  struct ScrubTabBehavior {
+    ScrubTabBehaviorType committed_info;
+    ScrubTabBehaviorType pending_info;
+  };
+
+  struct OpenTabParams {
```

Comment:
this doesn't seem to be used anywhere...?

Reply:
Thanks for noticing